com.unity.multiplayer.mlapi/Tests/Runtime/RpcPipelineTestComponent.cs

```diff
@@ -0,0 +1,196 @@
+using System.Collections.Generic;
+using UnityEngine;
+using MLAPI.Messaging;
+
+namespace MLAPI.RuntimeTests
+{
+    /// <summary>
+    /// Used in conjunction with the RpcQueueTest to validate:
+    /// - Sending and Receiving pipeline to validate that both sending and receiving pipelines are functioning properly.
+    /// - Usage of the ServerRpcParams.Send.UpdateStage and ClientRpcParams.Send.UpdateStage functionality.
+    /// - Rpcs receive will be invoked at the appropriate NetworkUpdateStage.
+    /// </summary>
+    public class RpcPipelineTestComponent : NetworkedBehaviour
+    {
+        /// <summary>
+        /// Allows the external RPCQueueTest to begin testing or stop it
+        /// </summary>
+        public bool PingSelfEnabled;
+
+        /// <summary>
+        /// How many times will we iterate through the various NetworkUpdateStage values?
+        /// (defaults to 2)
+        /// </summary>
+        public int MaxIterations = 2;
+
+
+        // Start is called before the first frame update
+        private void Start()
+        {
+            m_Serverparams.Send.UpdateStage = NetworkUpdateStage.Initialization;
+            m_Clientparams.Send.UpdateStage = NetworkUpdateStage.Update;
+
+            m_MaxStagesSent = (System.Enum.GetValues(typeof(NetworkUpdateStage)).Length)*MaxIterations;
+
+            //Start out with this being true (for first sequence)
+            m_ClientReceivedRpc = true;
+        }
+
+        /// <summary>
+        /// Determine if we have iterated over more than our maximum stages we want to test
+        /// </summary>
+        /// <returns>true or false (did we exceed the max iterations or not?)</returns>
+        public bool ExceededMaxIterations()
+        {
+            if(m_StagesSent.Count > m_MaxStagesSent && m_MaxStagesSent > 0)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns back whether the test has completed the total number of iterations
+        /// </summary>
+        /// <returns></returns>
+        public bool IsTestComplete()
+        {
+            if (m_Counter >= MaxIterations)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool m_ClientReceivedRpc;
+        private int m_Counter = 0;
+        private int m_MaxStagesSent = 0;
+        private ServerRpcParams m_Serverparams;
+        private ClientRpcParams m_Clientparams;
+        private NetworkUpdateStage m_LastUpdateStage;
+
+        // Update is called once per frame
+        private void Update()
+        {
+            if (NetworkingManager.Singleton.IsListening && PingSelfEnabled)
+            {
+                if (NetworkingManager.Singleton.IsListening && PingSelfEnabled && m_ClientReceivedRpc)
```

Comment:
I think this can just be 
```suggestion
                if (m_ClientReceivedRpc)
```

no?

Reply:
Not really.  Two things have to happen:

1. The host has to have started
2. The test has to have signaled that it is time to start testing.